Clients/Xamarin.Interactive.Client/ClientApp.cs

```diff
@@ -22,7 +22,7 @@
 
 namespace Xamarin.Interactive
 {
-    abstract class ClientApp
+    abstract class ClientApp : IDisposable
```

Comment:
`IDisposable` is not the right pattern for this. Just add a plain non-virtual `NotifyProcessWillTerminate` method or something and don't try to be clever with the pattern/`GC.SuppressFinalize`. If we need to make this virtual in the future we can do so.